src/TLuaInterpreter.cpp

```diff
@@ -1106,13 +1106,10 @@ int TLuaInterpreter::loadRawFile(lua_State* L)
                            "got %s!)",
                         luaL_typename(L, 1));
         return lua_error(L);
-    } else {
-        replayFileName = QString::fromUtf8(lua_tostring(L, 1));
-        if (replayFileName.isEmpty()) {
-            lua_pushnil(L);
-            lua_pushstring(L, "a blank string is not a valid replay file name");
-            return 2;
-        }
+    } else if (replayFileName = QString::fromUtf8(lua_tostring(L, 1)); replayFileName.isEmpty()) {
```

Comment:
:thinking: However we are in the situation that Clazy would report as an `else after return` here - and since we ***do*** want the `replayFileName` outside of the scope of the `else if` it is just as valid to have:

```
int TLuaInterpreter::loadRawFile(lua_State* L)
{
    if (!lua_isstring(L, 1)) {
        lua_pushfstring(L, "loadRawFile: bad argument #1 type (replay file name, {may include a relative to \n"
                           "profile's \"logs\" sub-directory, or an absolute path}, as string expected, \n"
                           "got %s!)",
                        luaL_typename(L, 1));
        return lua_error(L);
    }
    QString replayFileName{QString::fromUtf8(lua_tostring(L, 1))};
    if (replayFileName.isEmpty()) {
        lua_pushnil(L);
        lua_pushstring(L, "a blank string is not a valid replay file name");
        return 2;
    }

...
```